include/boost/gil/image_processing/fast_feature_detector.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef BOOST_GIL_IMAGE_PROCESSING_FAST_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_FAST_HPP
+
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/locator.hpp>
+#include <boost/gil/point.hpp>
+#include <algorithm>
+#include <cmath>
+#include <vector>
+namespace boost { namespace gil {
+namespace detail {
+template <typename srcview>
+bool fast_feature_detector(const srcview& buffer, int r, int c, std::vector<point_t>& points, int t)
+{
+    int valid_points_count = 16;
+    auto src_loc = buffer.xy_at(c, r);
+    std::vector<int> threshold_indicator, intensity_array(16);
+    std::vector<decltype(src_loc.cache_location(0, -1))> pointers(valid_points_count);
+    //stroring intensities of pixels on circumference beforehand to decrease runtime
+    for (int i = 0; i < 16; i++)
```

Comment:
Isn't `std::size_t` a better choice here?

Reply:
used std::size_t,std::ptrdiff_t wherever possible, size_t being unsigned in nature cannot be used for threshold_indicator